src/modules/Core/Wallets/api.js

```diff
@@ -82,21 +82,33 @@ export const getEnabledTokensFile = (wallet: AbcCurrencyWallet) => {
 }
 
 export async function setEnabledTokens (wallet: AbcCurrencyWallet, tokens: Array<string>, tokensToDisable?: Array<string>) {  // initialize array for eventual setting of file
-  let finalTextArray = tokens
+  let finalTextArray = [...tokens]
   // now stringify the new tokens
   let stringifiedTokens = JSON.stringify(finalTextArray)
   // grab the enabledTokensFile
+  const tokensFile = getEnabledTokensFile(wallet)
+  await tokensFile.setText(stringifiedTokens)
+  enableTokens(wallet, tokens)
+  if (tokensToDisable && tokensToDisable.length > 0) {
+    disableTokens(wallet, tokensToDisable)
+  }
+  return tokens
```

Comment:
Why even return tokens here as it's the same array that's passed into the function. The caller should already have access to it.